p/scripts/main.js

```diff
@@ -692,21 +692,113 @@ function init_posts() {
 	}
 }
 
+//Gets URL search parameter value.
+//Does not use URLSearchParams to ensure coverage of older versions of Internet Explorer.
+function get_search_param_value(param_name) {
+	const query_string = location.search;
+	const search_params_string = query_string.slice(1); //Removing '?' at '0' position.
+	const search_params = search_params_string.split('&');
+	let param_value = '';
+	let key_value_array;
+	search_params.forEach(function(key_value) {
+		key_value_array = key_value.split('=');
+		if (key_value_array[0] === param_name) {
+			param_value = key_value_array[1];
+			return;
+		}
+	});
+	return param_value;
+}
+
+//Adds category id to session storage when user opens corresponding category in the left sidebar.
+function add_category_to_session_storage(category_id) {
+	if ((category_id.charAt(0) !== 'c') && (category_id !== 'tags')) {
+		return;
+	}
+
+	let open_categories = sessionStorage.getItem('FreshRSS_open_categories');
+	if (!open_categories) {
+		open_categories = '';
+	}
+
+	if (open_categories.indexOf(category_id) === -1) {
+		open_categories += category_id + ',';
+	}
+	sessionStorage.setItem('FreshRSS_open_categories', open_categories);
+}
+
+//Removes category id from session storage when user closes corresponding category in the left sidebar.
+function remove_category_from_session_storage(category_id) {
+	if ((category_id.charAt(0) !== 'c') && (category_id !== 'tags')) {
+		return;
+	}
+
+	let open_categories = sessionStorage.getItem('FreshRSS_open_categories');
+	if (open_categories.indexOf(category_id) !== -1) {
```

Comment:
Wouldn't a JSON be easier to manipulate than a comma-separated string? Please give it a try.

Reply:
Sure, I'll make the update.